mountpoint-s3/src/prefetch/part_stream.rs

```diff
@@ -24,11 +26,12 @@ pub trait ObjectPartStream {
     /// size for the parts, but implementations are allowed to ignore it.
     fn spawn_get_object_request<Client>(
         &self,
-        client: &Client,
+        client: Arc<Client>,
```

Comment:
this change in the interface may not be required, but I'd like to make sure we agree on the memory limiting strategy first